src/Microsoft.Extensions.Configuration/ConfigurationBuilderExtensions.cs

```diff
@@ -7,9 +7,42 @@
 
 namespace Microsoft.Extensions.Configuration
 {
-    public static class MemoryConfigurationExtensions
+    public static class ConfigurationBuilderExtensions
     {
         /// <summary>
+        /// Includes an existing IConfiguration as a configuration provider to <paramref name="configuraton"/>.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="config">The <see cref="IConfiguration"/> to include.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddConfiguration(this IConfigurationBuilder configurationBuilder, IConfiguration configuration)
```

Comment:
That said, `Include()` and `Import()` seem decent candidates to me. So does `AddExistingConfiguration()`.

Note that we are not necessarily moving away from then pattern we use for the other providers, e.g. the name `AddChainedConfiguration()` I proposed happened to align with the name of the provider `ChainedConfigurationProvider` in this PR. If we don't like that maybe we still can fine a pair of nicer names for the method and the class that still align.